apps/calendar/js/views/event_base.js

```diff
@@ -193,30 +146,29 @@ EventBase.prototype = {
    *
    * @param {String} id busytime id.
    */
-  _loadModel: function(id, callback) {
+  _loadModel: co.wrap(function *(id) {
     var self = this;
     var token = ++this._changeToken;
     var classList = this.element.classList;
+    var result;
 
     classList.add(this.LOADING);
 
-    dayObserver.findAssociated(id).then(record => {
+    try {
+      var record = yield core.bridge.fetchRecord(id);
       if (token === self._changeToken) {
-        self.useModel(
-          record.busytime,
-          record.event,
-          callback
-        );
-      } else {
-        // ensure loading is removed
-        classList.remove(this.LOADING);
+        self._useModel(record);
       }
-    })
-    .catch(() => {
-      classList.remove(this.LOADING);
-      console.error('Error looking up records for id: ', id);
-    });
-  },
+    } catch(err) {
+      console.error(`Error looking up records for id: ${id}`);
+      console.error(`${err.message}\n${err.stack}`);
+      result = Promise.reject(err);
+    }
+
+    classList.remove(this.LOADING);
+
+    return result;
```

Comment:
in both cases we need to remove the loading class. `co.wrap` ensures that method returns a promise.. `return anyValue` is the *same* as `return Promise.resolve(anyValue)`